admin/main.js

```diff
@@ -1,85 +1,76 @@
-          headers: {
-            'Content-Type': 'application/json'
-          },
-          body: JSON.stringify(data)
-      }).then(response => {
-        response.json()
-        if (response.ok) {
-          location.reload(true);
-        };
-        console.log(response)
-        return response
-      } )     //  return response.json();
+    return fetch(url, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(data)
+    });
   }
-  fetch('/admin/rpc/list-users').then((response) => {
-    response.json().then(resp => {
-      for (const user of resp) {
-        const li = document.createElement('li');
-        const button = document.createElement('button');
-        const spanName = document.createElement('span');
-        const spanButton = document.createElement('span');
-        const strong = document.createElement('strong');
-        button.innerText ='x';
-        button.className = 'list'
-        spanButton.className = 'button'
-        spanName.innerText = user.name + ' ' + user.lastName;
-        li.appendChild(spanName);
-        if(username !== user.username) {
-          spanButton.appendChild(button);
-          li.appendChild(spanButton);
-          list.appendChild(li);
-          
-          button.addEventListener("click",  () => {
-            postData('/admin/rpc/delete-users', {
-                userID: user.userID
-            });
-          });
-        } else {
-          strong.innerText ='Admin';
-          spanButton.appendChild(strong);
-          li.appendChild(spanButton);
-          list.appendChild(li);
-          
-          button.addEventListener("click",  () => {
-            postData('/admin/rpc/delete-users', {
-                userID: user.userID
-            });
-          });
-        }
+
+  function refresh() {
+    list.innerHTML = '';
+    for (const user of users) {
+      const li = document.createElement('li');
+      const spanName = document.createElement('span');
+      const spanButton = document.createElement('span');
+      spanButton.className = 'button';
+      spanName.innerText = user.name + ' ' + user.lastName;
+      li.appendChild(spanName);
+      if (username !== user.username) {
+        spanButton.innerHTML = `<button class="list" data-id="${user.userID}">x</button>`;
+      } else {
+        spanButton.innerText = 'Admin';
       }
-    }, error => {
-      console.log(error);
-    });
+      li.appendChild(spanButton);
+      list.appendChild(li);
+    }
+  }
+
+  list.addEventListener('click',  (event) => {
+    const target = event.target;
+    if (target.classList.contains('list')) {
+      const userID = parseFloat(target.getAttribute('data-id'));
+      postData('/admin/rpc/delete-users', {
+        userID: userID
+      }).then(() => {
+        users = users.filter(user => userID !== user.userID);
+        refresh();
+      });
+    }
   });
+  
+  function reloadUsers() {
```

Comment:
This function just takes a fresh copy of the users from the database.
Note that in general we do not want to reload a page just because we added/removed/edited a user.